### PR TITLE
Implement training functions

### PR DESCRIPTION
**Pull request by MaHaWo**
- implement per-epoch training and test/evaluation functions 
- correct GNN-model -> single gnn-block to list thereof 
- add tests 

Questions: 
- Is this enough to work with hyperparameter optimization? 
- what else would you need?

### CODE REVIEW

File: QuantumGravPy/src/QuantumGrav/evaluate.py [hunk 1]
```diff
@@ -0,0 +1,176 @@
...
+    device: torch.device = torch.device("cpu"),
+    apply_model: Callable[[torch.nn.Module, torch_geometric.data.Data], Any] = None,
+) -> None:
+    """Train the model for one epoch. This will put the model into training mode, iterate over the data loader, compute the loss, and update the model parameters. It also processes the loss data if a processing function is provided or computes statistics on the loss data if not. Either output will be appended to the loss_data list.
+
+    Args:
+        model (torch.nn.Module): Model to train
+        data_loader (torch_geometric.data.DataLoader): Data loader for the training data
+        optimizer (torch.optim.Optimizer): Optimizer for updating the model parameters
+        criterion (Callable[[torch.Tensor, torch.Tensor], torch.Tensor  |  float]): Loss function to compute the loss
```
[the remark this answers is not in the document]
Comment (MaHaWo) [reply QuantumGravPy/src/QuantumGrav/evaluate.py:95]: no. this function, which is the loss function, is task specific. I would not put that here.
